### PR TITLE
WIP: Scala 3

### PR DESCRIPTION
**Pull request by ybasket**
Brings core, testkit & circe to Scala 3.0.0-RC2

Also resolves pretty much all Scala Steward PRs.

### CODE REVIEW

**Comment by ybasket** [comment]
@satabin All that can be done for Scala 3.0.0-RC2 is done and works (at least locally), so it's on you whether you want to merge now(-ish) or after Scala 3 has been released.

**Comment by satabin** [comment]
Is this now ready?

**Comment by ybasket** [comment]
> Is this now ready?

If Travis now accepts it, yes (just added Scala 3 to its matrix). Play JSON and Spray JSON wouldn't be published for Scala 3, only core and circe. Play JSON might follow soon-ish: https://github.com/playframework/play-json/pull/611

**Comment by satabin** [comment]
I did all the bulk updates, can you please rebase and resolve conflicts?

File: .travis.yml [hunk 1]
```diff
@@ -11,11 +11,12 @@ before_script:
 
 script:
   - export SBT_OPS=-Xmx4G
-  - if [[ -n "$CODACY_PROJECT_TOKEN" && $TRAVIS_SCALA_VERSION == 2.12* ]]; then sbt ++$TRAVIS_SCALA_VERSION clean coverage test coverageReport coverageAggregate codacyCoverage; else sbt ++$TRAVIS_SCALA_VERSION test; fi
+  - if [[ -n "$CODACY_PROJECT_TOKEN" && $TRAVIS_SCALA_VERSION == 2.12* ]]; then sbt ++$TRAVIS_SCALA_VERSION clean coverage test coverageReport coverageAggregate codacyCoverage; elif [[ $TRAVIS_SCALA_VERSION == 3.0* ]]; then sbt ++$TRAVIS_SCALA_VERSION coreJVM/test coreJS/test circeJVM/test circeJS/test testkitJVM/test testkitJS/test; else sbt ++$TRAVIS_SCALA_VERSION test; fi
```
Comment (ybasket) [inline .travis.yml:14]: This is ugly, but I don't see any other solution right now. set apparently doesn't properly handle if aggregations have different `crossScalaVersions` 😞: https://github.com/sbt/sbt/issues/3698

Once play-json is published against Scala 3 (PR exists), then one could think of dropping support for spray-json, the project seems pretty dead and users can still use diffson 4.0.3. Then all modules would be aligned again.

**Comment by ybasket** [comment]
@satabin ready for review now.

File: .travis.yml (hunk 1 above)
Comment (satabin) [reply .travis.yml:14]: I plan to switch to GH actions anyway, this will change.

**Pull request merged**